### PR TITLE
Support displaying relative change in statistics graph cards

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Support displaying relative change in statistics graph cards

Absolute value (already supported, but will use the `state` instead of the `sum` with this PR):
image

Relative change during a period (added by this PR):
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/recorder.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { computeStateName } from "../common/entity/compute_state_name";
 import { HomeAssistant } from "../types";
 
-export type StatisticType = "sum" | "min" | "max" | "mean";
+export type StatisticType = "sum" | "sum_rel" | "min" | "max" | "mean";
```
Comment (bramkragten) [inline src/data/recorder.ts:4]: `sum_rel` is not a `StatisticType`?

File: src/components/chart/statistics-chart.ts [hunk 2]
```diff
@@ -346,9 +346,9 @@ class StatisticsChart extends LitElement {
         const dataValues: Array<number | null> = [];
         statTypes.forEach((type) => {
           let val: number | null;
-          if (type === "sum") {
+          if (type === "sum" || type === "sum_rel") {
             if (initVal === null) {
-              initVal = val = stat.state || 0;
+              initVal = val = type === "sum_rel" ? 0 : stat.sum || 0;
```
Comment (bramkragten) [inline src/components/chart/statistics-chart.ts:351]: Should this be `stat.sum` or `stat.state`? We want the total right, that would be `stat.state`, as `stat.sum` could have been reset somewhere?

File: src/data/recorder.ts [hunk 3]
```diff
@@ -201,10 +201,13 @@ export const calculateStatisticsSumGrowth = (
 export const statisticsHaveType = (
   stats: StatisticValue[],
   type: StatisticType
-) => stats.some((stat) => stat[type] !== null);
+) => {
+  type = type === "sum_rel" ? "sum" : type;
```
Comment (bramkragten) [inline src/data/recorder.ts:205]: I think this should be handled in `statistics-chart.ts`

File: src/components/chart/statistics-chart.ts (hunk 2 above)
Comment (emontnemery) [reply src/components/chart/statistics-chart.ts:351]: It should be `stat.sum`.
The `stat.state` is just used as a scratch pad for sensor to remember its state in the previous period, and we should perhaps remove it from the WS API response.

Comment (emontnemery) [reply src/components/chart/statistics-chart.ts]: We discussed this a bit further, and the conclusion was we should in fact display `stat.state` as is.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**